src/openforms/formio/dynamic_config/__init__.py

```diff
@@ -44,7 +45,25 @@ def rewrite_formio_components_for_request(
     Loop over the formio configuration and inject request-specific configuration.
     """
     for component in configuration_wrapper:
-        register.update_config_for_request(component, request=request)
+        # prevent multiple overwrites of content components (bandaid for #2895)
+        # postprocess (a) non-content components, (b) components with inline style
+        # (c) components with nonce that does not match the nonce on the request
+
+        # get HTML (if any) from component and parse it
+        if html_string := component.get("html", None):
+            request_nonce = request.headers.get("X-Csp-Nonce", None)
+            html = lxml.html.fromstring(html_string)
+            tags = list(html.iter())
+            tags_with_inline_style = [tag for tag in tags if tag.get("style")]
+            tags_with_nonce = [tag for tag in tags if tag.get("nonce")]
+            html_nonce = tags_with_nonce[0].get("nonce") if tags_with_nonce else None
+
+        if component["type"] != "content":
+            register.update_config_for_request(component, request=request)
+        elif tags_with_inline_style:
+            register.update_config_for_request(component, request=request)
+        elif request_nonce and html_nonce and html_nonce != request_nonce:
+            register.update_config_for_request(component, request=request)
```

Comment:
Can you move this patch into the correct place - the Formio component plugin at `src/openforms/formio/components/vanilla.py`, in the `Content` class